### PR TITLE
Expand GraphQL sort/filter on stocktake, requisition, invoice lines

### PR DESCRIPTION
**Pull request by andreievg**
# 👩🏻‍💻 What does this PR do?

This PR extends server-side sorting and filtering across three areas of the GraphQL API, grounded in what the frontend currently displays/sorts/filters (client-side only, in most cases) rather than speculative additions:

**1. `stocktakeLines`** — adds sort + filter fields that were previously client-side only:
- Sort: `manufactureDate`, `donor`, `campaign` (campaign required a new `left_join` to the `campaign` table — the only new join added in this PR; everything else reuses tables already joined in the repository query)
- Filter: `donorId`, `campaignId`, `campaignName`, `batch` (string/contains), `manufactureDate` (date range)

**2. `invoiceLines`** — closes gaps where a field was already sortable but had no matching filter:
- Filter: `batch`, `expiryDate`, `locationCode`, `manufactureDate` (the latter also gets its first sort field)

**3. New `requisitionLines` top-level query** — mirrors the existing `stocktakeLines` query shape (didn't exist before; requisition lines were only reachable nested under `requisition { lines { ... } }` with no filter/sort/pagination args). Supports:
- Sort: `itemCode`, `itemName`, `requestedQuantity`, `suggestedQuantity`, `supplyQuantity`, `approvedQuantity`, `comment`, `defaultPackSize`, `monthsOfStock`
- Filter: `id`, `storeId`, `requisitionId`, `itemId`, `itemCodeOrName`, `requestedQuantity`, `comment`, `monthsOfStock`

**4. `items(notInRecord: ...)` filter** — new filter that excludes items already present as a line on a given stocktake, requisition, or invoice:
```graphql
notInRecord: { record: STOCKTAKE | REQUISITION | INVOICE, id: "<record id>" }
```
Implemented as a `NOT IN (subquery)` against the relevant line table's `item_id`, mirroring the existing `masterListId`/`isProgramItem` subquery pattern in `ItemFilter` — no new joins needed.

**5. `monthsOfStock` filter/sort on `requisitionLines`** — new SQL view (`requisition_line_months_of_stock`) computing `available_stock_on_hand / average_monthly_consumption` per requisition line, exposed as a new generic `FloatFilter` input (`equalTo` / `belowOrEqualTo` / `aboveOrEqualTo` / `belowThan` / `aboveThan`) and a matching sort field.

## How the frontend would use these (no frontend changes are in this PR)

**Item search — `notInRecord`:**

Every "add item" modal across stocktake/requisition/invoice lines (`StocktakeLineEditForm`, `RequestLineEdit`, `ResponseLineEdit`, `SelectItem` for outbound shipments, `InboundLineEditForm`, the supplier/customer return `ItemSelector`s, `PrescriptionLineEdit`) currently excludes already-added items via a client-side `id: { notEqualAll: existingItemIds }`, where `existingItemIds` is built by mapping over the full, already-loaded line list in memory. That means correctness depends on the client having *all* lines loaded up front — it doesn't scale to a paginated line list and duplicates the same "existing item ids" computation in ~8 places.

With this PR, those call sites can instead pass:
```graphql
items(filter: { notInRecord: { record: STOCKTAKE, id: stocktakeId } }) { ... }
```
and drop the client-side `existingItemIds` mapping and the `notEqualAll` filter entirely — the exclusion is computed server-side in a single query, independent of how many lines are already loaded or whether they're paginated.

**"Hide over-stocked" toggle (Request Requisition) — `monthsOfStock`:**

Today `useRequestLines.ts` filters the requisition's already-loaded lines in JS: for each line, `passesFilter = (SOH == 0 && AMC == 0) || SOH < AMC * threshold`, where `threshold = minMonthsOfStock ?? maxMonthsOfStock` (both plain fields on the requisition). This only works because Request-requisition `itemStats` are a frozen snapshot baked directly onto `requisition_line` at generation time (not a live computation like Response requisitions), so the values were already available client-side without an extra fetch.

With this PR, the same comparison can be pushed server-side once the frontend switches from the nested `requisition.lines` field to the new `requisitionLines` query:
```graphql
requisitionLines(filter: { monthsOfStock: { belowThan: <threshold> } })
```
The view's `monthsOfStock` column already reproduces the client's edge cases: `0` when both stock-on-hand and consumption are zero (matches the client's explicit special case), and `NULL` when consumption is zero but stock is present (naturally excluded by any threshold filter, matching the client never treating a zero-consumption/nonzero-stock line as "over-stocked"). The `belowThan`/`aboveThan` variants are strict (`<`/`>`), matching the client's existing strict comparison exactly (as opposed to `belowOrEqualTo`/`aboveOrEqualTo`, which are inclusive) — no behavioural drift at the exact threshold boundary.

The one piece that stays client-side by necessity: the toggle's "always show the currently-open draft line" exception is UI-only state (which row is open in an edit modal) with no server-side equivalent — it should stay as a small client-side OR-in after the filtered list comes back.

## 💌 Any notes for the reviewer?

- No frontend changes are included — this is additive, backend-only API surface. The two usage patterns above (`notInRecord`, `monthsOfStock`) are proposals for follow-up frontend PRs, not implemented here.
- The `campaign` join on `stocktakeLines` is the only new join added anywhere in this PR; everything else reuses existing joins. Adding it required a new `joinable!`/`allow_tables_to_appear_in_same_query!` pair (`stocktake_line` ↔ `campaign`, plus `campaign` ↔ `reason_option` which diesel required transitively).
- `requisition_line_months_of_stock` is a new SQL view registered in the existing view-migration registry (`server/repository/src/migrations/views/mod.rs`), positioned after `link_views` since it depends on `requisition_line_view`.
- `FloatFilter`/`FloatFilterInput` are new generic filter types (mirroring the existing `DateFilter` shape) since no existing filter type supported one-sided numeric threshold comparisons — reusable for future numeric range filters elsewhere in the schema.
- Existing `RequisitionLineFilter`/`RequisitionLineSortField`/`StocktakeLineFilter` behavior is unchanged for all pre-existing fields; this is purely additive.

# 🧪 Tested

- Full server workspace build (`cargo build --features sqlite`) passes clean, including all binary targets (server, cli, android, windows).
- Full server test suite (`cargo nextest run --features sqlite`) passes: 1333/1333 tests.
- Added a new GraphQL test for the `requisitionLines` query (`graphql_requisition_line::requisition_line_queries::test::test_graphql_requisition_lines_query`), exercised end-to-end through the mock service provider and item dataloader.
- Ran the view-migration round-trip test (`repository::migrations::views::test::drop_and_rebuild_views`) to confirm the new `requisition_line_months_of_stock` view's SQL (including the `CASE WHEN`/`NULL` edge cases) is valid on the compiled backend.
- Manually traced the `months_of_stock` computation against the existing client-side "hide over-stocked" logic (`useRequestLines.ts`) field-by-field to confirm the `NULL`/`0` edge-case handling and strict `belowThan`/`aboveThan` comparisons reproduce it exactly.

# 📃 Documentation

- [x] **No documentation required** — no user-facing change (backend API surface only; no frontend consumes it yet)

### CODE REVIEW

File: server/repository/src/db_diesel/filter_sort_pagination.rs [hunk 1]
```diff
@@ -295,6 +295,59 @@ impl DateFilter {
...
+pub struct FloatFilter {
+    #[ts(optional)]
+    pub equal_to: Option<f64>,
+    #[ts(optional)]
+    pub below_or_equal_to: Option<f64>,
+    #[ts(optional)]
+    pub above_or_equal_to: Option<f64>,
+    /// Strict less-than (exclusive)
+    #[ts(optional)]
+    pub below_than: Option<f64>,
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/filter_sort_pagination.rs:308]: This is really nit, but below_than and above_than isn't proper English XD, would have just named less_than and greater_than 👀

File: server/repository/src/db_diesel/requisition_line/requisition_line_row.rs [hunk 2]
```diff
@@ -57,9 +57,26 @@ define_linked_tables! {
...
-    TS, Clone, Queryable, Debug, PartialEq, Default, Serialize, Deserialize,
-)]
+table! {
+    requisition_line_months_of_stock (requisition_line_id) {
+        requisition_line_id -> Text,
+        months_of_stock -> Nullable<Double>,
+    }
+}
+
+#[derive(Clone, Insertable, Queryable, Debug, PartialEq, Default)]
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/requisition_line/requisition_line_row.rs:67]: This is a view, no? Cannot be inserted

```suggestion
#[derive(Clone, Queryable, Debug, PartialEq, Default)]
```

**Review by roxy-dao — CHANGES_REQUESTED** [review]
Thanks for the changes Andrei~

Just a few things changes below~
- Would be nice to have tests for the new SQL~? 
- campaignName should be a string filter not an equal filter. We do the same for master list name
- RequisitionLineFilterInput.store_id can probably be removed since we're using 
```js
    filter.store_id = Some(EqualFilter::equal_to(store_id.to_string()));
```

**Comment by jmbrunskill** [comment]
closing, re-opened in internal repo
